### ISSUE TITLE
Add Prod Documentation on Dockerhub

### ISSUE THREAD

**Issue by JohannesFleischer** [hashicorp/vault#21494]
**Is your feature request related to a problem? Please describe.**
I am currently starting with vault and boundary, and I am reading the docs to set up first test instances of the services with docker-compose.
Besides the fact that there is not really much information about the usage with docker in the docs itself, the information found on [Dockerhub](https://hub.docker.com/r/hashicorp/vault) only includes information about the dev mode. So it took me quite a while to realize that it is not possible to start the official image in prod mode without overwriting the default command that is hard-coded in the Dockerfile itself with:
```yml
    command: "server"
```

**Describe the solution you'd like**
Information added on how to use the official container in prod mode on Dockerhub.
Furthermore, docs about using the project with docker-compose would be nice.

**Comment by Malshtur** [comment]
I agree the doc could be a bit richer. What could be nice to find would be :

- docker-compose example
- a using raft storage with mounted volume if possible as file storage backend should not be used in production
- TLS enabled

May be I can try to include what I use and we could go from there and improve it to doc grade ? I am by no means an expert though.

**Comment by JohannesFleischer** [comment]
@Malshtur yeah, why not. If you want, you can of course send your files, and we can take a look at them.
But I'm also not experienced with this tool.

**Comment by hsimon-hashicorp** [comment]
@yhyakuna Is this something your team can help with?

**Comment by ssstroke** [comment]
Bump
